src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -88,103 +128,1169 @@ static DamageTable D_80A31B1C = {
+ * Update actor scale for slime effect.
+ * If collision, knockback actor. If actor has stopped moving
+ * in Y direction, initialize landing action.
+ */
+void EnSlime_UpdateJump(EnSlime* this, PlayState* play) {
+    this->timer--;
+    if (this->collider.base.atFlags & AT_HIT) {
+        if (this->actor.speed > 0.0f) {
+            this->actor.speed *= -1.2f;
+            this->collider.base.atFlags &= ~AT_HIT;
+        }
+    }
+    Math_StepToF(&this->actor.scale.x, 0.008f, 0.0025f);
+    Math_StepToF(&this->actor.scale.y, 0.011f, 0.0025f);
+    this->actor.scale.z = this->actor.scale.x;
+    if ((this->actor.velocity.y < 0.0f) && (this->actor.bgCheckFlags & 1)) {
+        EnSlime_FinishJump(this);
+    }
+}
+
+/**
+ * Initialize jump landing.
+ * Adjust timer and scale, and stop xz movement.
+ * Set action function to UpdateJumpLanding
+ */
+void EnSlime_FinishJump(EnSlime* this) {
+    this->timer = 15;
+    this->actor.scale.x = 0.0132f;
+    this->actor.scale.y = 0.0074399994f;
+    this->actor.scale.z = 0.0132f;
+    this->actionFunc = EnSlime_UpdateJumpLanding;
+    this->actor.speed = 0.0f;
+}
+
+/**
+ * Update scale/rot for ooze jiggle effect (XZ), update Y scale
+ * to make it appear as though chuchu is compressing.
+ * If > 12 frames have passed since jump start, check again if player
+ * is in range and restart engage/jump cycle if so.
+ * If too far from home, initialize return home cycle (takes priority over
+ * player engage check).
+ * Otherwise, if timer is up, return to idle.
+ */
+void EnSlime_UpdateJumpLanding(EnSlime* this, PlayState* play) {
+    f32 factorScaleY;
+    f32 factorRotXZ;
+    f32 factorScaleXZ;
+
+    EnSlime_UpdateBlink(this);
+    this->timer--;
+    factorScaleY = ((this->timer / 5) + 1) * 1.6f;
+    factorRotXZ = sqrtf(this->timer) * 0.2f;
+    factorScaleXZ = ((cos_rad(this->timer * (2.0f * M_PI / 5.0f)) * (0.05f * factorScaleY)) + 1.0f) * 0.01f;
+    this->actor.scale.x = factorScaleXZ;
+    this->actor.scale.z = factorScaleXZ;
+    if (this->timer < 15) {
+        this->actor.scale.y = (1.0f - (sin_rad(this->timer * (2.0f * M_PI / 5.0f)) * (0.04f * factorScaleY))) * 0.01f;
+    }
+    this->actor.shape.rot.x = randPlusMinusPoint5Scaled(0x200) * factorRotXZ;
+    this->actor.shape.rot.z = randPlusMinusPoint5Scaled(0x200) * factorRotXZ;
+
+    if (Actor_WorldDistXZToPoint(&this->actor, &this->actor.home.pos) > this->distLimit) {
+        EnSlime_ReturnHome(this);
+    } else if ((Player_GetMask(play) != PLAYER_MASK_STONE) && (this->actor.xzDistToPlayer < 280.0f) &&
+               (this->timer < 12)) {
+        EnSlime_EngagePlayer(this);
+    } else if (this->timer == 0) {
+        EnSlime_ReturnToIdle(this);
+    }
+}
+
+/**
+ * Initialize reaction to receiving a blunt hit (such as with Goron Punch etc.)
+ * Chuchu does not take damage, but appears slide backwards.
+ * Action function set to UpdateBluntHitReaction.
+ */
+void EnSlime_ReactToBluntHit(EnSlime* this) {
```

Comment:
This is another setup function; I think this one can be `EnSlime_SetupReactToBluntHit`, and then the action function can be `EnSlime_ReactToBluntHit`.